data/scrape/scrape.py

```diff
@@ -116,23 +116,38 @@ def _build_scraping_config(
 def main(args: argparse.Namespace) -> None:
     with open(LANGUAGES_PATH, "r", encoding="utf-8") as source:
         languages = json.load(source)
-    codes = list(languages.keys())
-    # Verifies language code for --restriction is valid.
+    codes = frozenset(languages.keys())
     if args.restriction:
         # Cleans entry.
-        keys = re.split(r"[;,\s]+\s*", args.restriction[0].strip(";, "))
-        if not keys[0]:
+        restriction_set = frozenset(
+            re.split(r"[;,\s]+\s*", args.restriction.strip(";, "))
+        )
+        if len(restriction_set) == 1 and not list(restriction_set)[0]:
             # Checks for empty entry.
             logging.fatal("Restriction flag raised but no language provided")
             exit(1)
-        rset = frozenset(args.restriction)
```

Comment:
This line was the bug that I was referring to offline. It should've been `rset = frozenset(keys)` to honor what the documentation suggests (i.e., the readme example with `--restriction='lit; spa'`).

As I continued to add the `--exclude` flag below, I refactored the code around here as well, which might have obscured what would otherwise be a one-line bug fix...